rust/rope/src/rope.rs

```diff
@@ -260,7 +260,7 @@ impl Metric<RopeInfo> for LinesMetric {
     }
 
     fn prev(s: &String, offset: usize) -> Option<usize> {
-        memrchr(b'\n', &s.as_bytes()[..offset]).map(|pos| pos + 1)
+        memrchr(b'\n', &s.as_bytes()[..offset - 1]).map(|pos| pos + 1)
```

Comment:
It might be worth adding something like, `debug_assert!( offset > 0, "caller is responsible for validating input")`, as a way of documenting our invariants?